git-pull-request

```diff
@@ -50,7 +50,22 @@ repo.git.remote('prune', 'origin')
 try:
   diff = repo.git.diff(local_branch, remote_ref)
 except git.exc.GitCommandError:
-  util.fatal('Remote branch %s not available.' % remote_ref)
+  push_it = util.prompt('Remote branch %s not available; attempt to push it up? [Y/n]' % remote_ref)
+  if push_it.lower() in ['y', '']:
+    split_remote_ref = remote_ref.split('/', 1)
+    if len(split_remote_ref) != 2:
+      util.fatal('Remote ref name "%s" in unexpected format (expected '
+        '"[origin]/[branchname]"' % remote_ref)
+      exit(1)
+    repo.git.push([split_remote_ref[0], split_remote_ref[1]])
+    try:
+      # Try again to diff against remote branch
+      diff = repo.git.diff(local_branch, remote_ref)
+    except git.exc.GitCommandError:
+      util.info('Remote branch %s is STILL not available, nothing to do.' % remote_ref)
```

Comment:
Should we just exit here without falling through to the error on 71?

Reply:
good call, fixed